core/src/parquet/util/hash_util.rs

```diff
@@ -45,17 +47,19 @@ const MURMUR_R: i32 = 47;
 unsafe fn murmur_hash2_64a(data_bytes: &[u8], seed: u64) -> u64 {
     let len = data_bytes.len();
     let len_64 = (len / 8) * 8;
-    let data_bytes_64 =
-        std::slice::from_raw_parts(&data_bytes[0..len_64] as *const [u8] as *const u64, len / 8);
```

Comment:
This is unsafe because it does not create a u64-aligned buffer and results in a panic with Rust 1.78